File: #206.Reverse Linked List/Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head==NULL||head->next==NULL){
            return head;
        }
        ListNode* p=head->next;
        ListNode* temp;
        head->next=NULL;
        while(p!=NULL){
            temp=p->next;
            p->next=head;
            head=p;
            p=temp;
        }
        return head;
    }
};

File: #206.Reverse Linked List/Reverse Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {

    if(head==null){
        return head;
    }

    var newHead=head;
    head=newHead.next;
    newHead.next=null;

    while(head){
        var temp=head.next;
        head.next=newHead;
        newHead=head;
        head=temp;
    }

    return newHead;
};